pkg/bpf/map_linux_test.go

```diff
@@ -51,64 +43,55 @@ func (k *TestKey) New() MapKey    { return &TestKey{} }
 func (v *TestValue) String() string { return fmt.Sprintf("value=%d", v.Value) }
 func (v *TestValue) New() MapValue  { return &TestValue{} }
 
-var _ = Suite(&BPFPrivilegedTestSuite{})
-
-func (s *BPFPrivilegedTestSuite) SetUpSuite(c *C) {
-	testutils.PrivilegedTest(c)
+func setup(tb testing.TB) {
+	testutils.PrivilegedTest(tb)
 
 	CheckOrMountFS("")
 
 	if err := rlimit.RemoveMemlock(); err != nil {
-		c.Fatal(err)
+		tb.Fatal(err)
 	}
 
+	testMap = NewMap("cilium_test",
+		ebpf.Hash,
+		&TestKey{},
+		&TestValue{},
+		maxEntries,
+		BPF_F_NO_PREALLOC,
+	).WithCache()
+
 	if err := testMap.OpenOrCreate(); err != nil {
-		c.Fatal("Failed to create map:", err)
+		tb.Fatal("Failed to create map:", err)
 	}
 
-	s.teardown = func() error {
-		testMap.Close()
+	tb.Cleanup(func() {
+		err := testMap.Close()
+		require.NoError(tb, err)
 
 		path, err := testMap.Path()
-		if err != nil {
-			return err
-		}
-
-		return os.Remove(path)
-	}
-}
-
-func (s *BPFPrivilegedTestSuite) TearDownSuite(c *C) {
-	if s.teardown != nil {
-		if err := s.teardown(); err != nil {
-			c.Fatal(err)
-		}
-	}
+		require.NoError(tb, err)
+		require.NoError(tb, os.Remove(path))
+	})
 }
 
 var (
 	maxEntries = 16
-
-	testMap = NewMap("cilium_test",
-		ebpf.Hash,
-		&TestKey{},
-		&TestValue{},
-		maxEntries,
-		BPF_F_NO_PREALLOC,
-	).WithCache()
+	testMap    *Map
```

Comment:
See my comments above, we should remove this one. :slightly_smiling_face:

Reply:
thanks, this should be addressed in https://github.com/cilium/cilium/pull/32220